docs/DeveloperGuide.md

```diff
@@ -1,34 +1,845 @@
+
+> **Warning:**
+> If `IndexOutOfBoundsException` or `NumberFormatException` is caught, a new `IncorrectCommand` class will be called to print the respective error messages
+
+**Step 3**  
+A new instance of `DeleteCommand` with `delete index` initialized will be created. The `execute()` method of `DeleteCommand` will then be called.
+
+**Step 4**  
+The `execute()` method will do 2 things:
+
+-   If there are no tasks in the existing task list, it will initialize a new `CommandResult` class that prints out an error message indicating an empty task list
+
+-   If there are tasks in the existing task list, the `DeleteCommand` class will call the `deleteTask()` method from the `TaskList` class to delete the task, based on the index. At the end of the execution, the `DeleteCommand` class will initialize a new `CommandResult` class that prints out the success message for task deletion.
+
+The following sequence diagram summarizes how delete command operation works:  
+
+![delete task](images/delete.png)
+
+#### 4.1.1. Design Considerations
+
+-   Calling `remove()` method in `deleteTask()` command of `TaskList` method instead of calling `remove()` method within the `execute()` method of the `DeleteCommand` class
+
+    -   Pros: Easier implementation for other classes that requires the same use.
+
+    -   Cons: Increased coupling amongst classes, which makes it harder for testing.
+
+    -   Rationale: We decided to implement it in such a way because we feel that the effects of increased coupling in such a case is minimal and testing for related classes and methods are not affected much. Furthermore, such implementation also allows us to keep all the related commands to the list of tasks within a class which keeps our code cleaner.
+
+### 4.2. Search task feature
+#### 4.2.1 Current Implementation
+
+The search task feature is currently implemented in both `SearchCommand` class and `SearchdCommand` class. Both classes inherit from the `Command` class.
+
+-   `SearchCommand` initializes the `taskType` to check which tasks the search function to search from and `searchParam` to get the search query that the user inputs.
+
+-   Similar to the `SearchCommand`, `SearchdCommand` initializes `taskType` to check the tasks that the search function has to search through and `searchParam` to get the search query that the user inputs. It also has a `date` parameter to check the date that the users wants to search from
+
+Given below is an example usage of the `Search` command:  
+
+**Step 1**  
+The user launches the app and retrieves the tasks that are saved under a local file using `Storage`.
+
+**Step 2**  
+The user enters `search t\{TASK TYPE} n\{SEARCH QUERY}` into the command line. Method `parseCommand()` from the `Parser` class will be called to parse the command provided.
+
+**Step 3**  
```

Comment:
Maybe can break down to more steps? This step is very complicated and hard to digest.